spyder/plugins/updatemanager/widgets/update.py

```diff
@@ -622,6 +677,51 @@ def __init__(self, icon=None, text=None, parent=None):
         self.setTextFormat(Qt.RichText)
 
 
+class DetailedUpdateMessageBox(UpdateMessageBox, SpyderFontsMixin):
+    def __init__(self, icon=None, text=None, parent=None, details=None):
+        super().__init__(icon=icon, text=text, parent=parent)
+        self.setSizeGripEnabled(True)
+        self.details = None
+        self.setDetailedText(details)
+
+    def setDetailedText(self, details=None):
+        """
+        Override setDetailedText.
+
+        Note: It is critical that QGridLayout.setRowStretch is called after
+        QMessageBox.setDetailedText in order for the stretch behavior to work
+        properly. That is the primary reason for overriding setDetailedText.
+        """
+        if self.details is not None:
+            self.details.setText(details)
+            return
+
+        super().setDetailedText(details)
+        self.details = self.findChild(QTextEdit)
+
+        self.details.setFont(self.get_font(SpyderFontType.Monospace))
+        self.details.setLineWrapMode(self.details.NoWrap)
+        self.details.setMinimumSize(400, 110)
+        self.details.setLineWrapMode(0)
+
+        qgl = self.findChild(QGridLayout)
+        qgl.setRowStretch(1, 0)
+        qgl.setRowStretch(3, 100)  # QTextEdit should take all the stretch
+
+    def event(self, event):
+        if event.type() in (event.LayoutRequest, event.Resize):
+            if event.type() == event.Resize:
+                result = super().event(event)
+            else:
+                result = False
+            if self.details is not None:
```

Comment:
```suggestion
            
            # Allow resize only if details is available and visible.
            if self.details and self.details.isVisible():
```
slight improvement, see above as well